FFmpegInterop/SampleProvider.cpp

```diff
@@ -61,8 +61,11 @@ namespace winrt::FFmpegInterop::implementation
 			}
 
 			MediaPropertySet properties{ audioEncProp.Properties() };
-			properties.Insert(MF_MT_AUDIO_CHANNEL_MASK, PropertyValue::CreateUInt32(
-				codecPar->channel_layout != 0 ? static_cast<uint32_t>(codecPar->channel_layout) : static_cast<uint32_t>(av_get_default_channel_layout(codecPar->channels))));
+
+			if (codecPar->ch_layout.order == AV_CHANNEL_ORDER_NATIVE)
+			{
+				properties.Insert(MF_MT_AUDIO_CHANNEL_MASK, PropertyValue::CreateUInt32(static_cast<uint32_t>(codecPar->ch_layout.u.mask)));
+			}

```

Comment:
If we see one of the other orders (CUSTOM, AMBISONIC, dunno if there are others) should we throw an error? Looks like we'll get undefined results currently.

Reply:
I've only seen AV_CHANNEL_ORDER_UNSPEC and AV_CHANNEL_ORDER_NATIVE in my testing. I'm ok with throwing an error if we see AV_CHANNEL_ORDER_CUSTOM or AV_CHANNEL_ORDER_AMBISONIC, since those scenarios haven't been validated.